### PR TITLE
refactor: update `blas/ext/base/dnannsumkbn2` to follow current project…

### LINKED ISSUE #1480: [RFC]: refactor `blas/ext/base/dnannsumkbn2` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dnannsumkbn2`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dnannsumkbn2) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by vr-varad**
## Description

> What is the purpose of this pull request?

refactor: update blas/ext/base/dnannsumkbn2 to follow current project conventions

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1480 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by vr-varad** [comment]
@kgryte I am getting errors in running tests and constantly getting error: 
Running test: lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/test/test.dnannsumkbn2.js

29 tests complete (38ms)

Running test: lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/test/test.dnannsumkbn2.native.js
/home/runner/work/stdlib/stdlib/lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/lib/dnannsumkbn2.native.js:48
	addon( N, x, strideX, out, strideOut );
	^

RangeError: invalid argument. Fourth argument has insufficient elements based on the associated stride and the number of indexed elements.
    at dnannsumkbn2 (/home/runner/work/stdlib/stdlib/lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/lib/dnannsumkbn2.native.js:48:2)
    at Test.test (/home/runner/work/stdlib/stdlib/lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/test/test.dnannsumkbn2.native.js:59:6)
    at Test.bound [as _cb] (/home/runner/work/stdlib/stdlib/node_modules/tape/lib/test.js:58:32)
    at Test.run (/home/runner/work/stdlib/stdlib/node_modules/tape/lib/test.js:77:10)
    at Test.bound [as run] (/home/runner/work/stdlib/stdlib/node_modules/tape/lib/test.js:58:32)
    at Immediate.next (/home/runner/work/stdlib/stdlib/node_modules/tape/lib/results.js:70:15)
    at processImmediate (node:internal/timers:466:21)
   

can u help me with this?

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/benchmark/benchmark.js [hunk 1]
```diff
@@ -41,16 +43,14 @@ var dnannsumkbn2 = require( './../lib/dnannsumkbn2.js' );
...
-	x = new Float64Array( len );
-	for ( i = 0; i < len; i++ ) {
-		if ( randu() < 0.2 ) {
-			x[ i ] = NaN;
-		} else {
-			x[ i ] = ( randu()*10.0 ) - 20.0;
+	function clbk() {
+		if ( bernoulli( 0.7 ) > 0 ) {
+			return discreteUniform( -10, 10 );
 		}
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/benchmark/benchmark.js:50]:
```suggestion
				return NaN;
		}
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/docs/repl.txt [hunk 2]
```diff
@@ -46,26 +46,25 @@
...
+
 {{alias}}.ndarray( N, x, strideX, offsetX, out, strideOut, offsetOut )
     Computes the sum of double-precision floating-point strided array elements,
     ignoring `NaN` values and using a second-order iterative Kahan–Babuška
     algorithm and alternative indexing semantics.
 
     While typed array views mandate a view offset based on the underlying
-    buffer, the `offset` parameter supports indexing semantics based on a
-    starting index.
+    buffer, the offsetX and offsetY parameter supports indexing semantics
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/docs/repl.txt]:
```suggestion
    buffer, the offset parameters supports indexing semantics
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/lib/dnannsumkbn2.native.js [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
 // MODULES //
 
-var addon = require( './../src/addon.node' );
+var addon = require( './../src/addon.node');
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/lib/dnannsumkbn2.native.js]:
```suggestion
var addon = require( './../src/addon.node' );
```

**Comment by vr-varad** [comment]
@kgryte @Planeshifter  I am getting errors in running tests and constantly getting error: 
Running test: lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/test/test.dnannsumkbn2.js

29 tests complete (38ms)

Running test: lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/test/test.dnannsumkbn2.native.js
/home/runner/work/stdlib/stdlib/lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/lib/dnannsumkbn2.native.js:48
	addon( N, x, strideX, out, strideOut );
	^

RangeError: invalid argument. Fourth argument has insufficient elements based on the associated stride and the number of indexed elements.
    at dnannsumkbn2 (/home/runner/work/stdlib/stdlib/lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/lib/dnannsumkbn2.native.js:48:2)
    at Test.test (/home/runner/work/stdlib/stdlib/lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/test/test.dnannsumkbn2.native.js:59:6)
    at Test.bound [as _cb] (/home/runner/work/stdlib/stdlib/node_modules/tape/lib/test.js:58:32)
    at Test.run (/home/runner/work/stdlib/stdlib/node_modules/tape/lib/test.js:77:10)
    at Test.bound [as run] (/home/runner/work/stdlib/stdlib/node_modules/tape/lib/test.js:58:32)
    at Immediate.next (/home/runner/work/stdlib/stdlib/node_modules/tape/lib/results.js:70:15)
    at processImmediate (node:internal/timers:466:21)
   

can u help me with this?

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/src/addon.c [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+/**
+* Receives JavaScript callback invocation data.
+*
+* @private
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon(napi_env env, napi_callback_info info) {
+    STDLIB_NAPI_ARGV(env, info, argv, argc, 5);
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/src/addon.c]: @vr-varad At this point you've contributed multiple PRs, and I'd expect that you'd have picked up on indentation practices. This file should be TABS, not spaces. Additionally, you are not adhering to our spacing conventions around arguments, etc.

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/src/addon.c [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    int64_t io;
+    if (strideOut < 0) {
+        io = -strideOut;
+    } else {
+        io = 0;
+    }
+
+    double *out = (double *)Out;
+    int64_t n;
+    out[io] = stdlib_strided_dnannsumkbn2(N, (double *)X, strideX, &n);
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/src/addon.c]:
```suggestion
    out[io] = stdlib_strided_dnannsumkbn2(N, X, strideX, &n);
```

Please read up on the documentation and the source code in the various macro packages. This explicit cast is not necessary here. Same thing for `Out` above.

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/lib/ndarray.native.js [hunk 6]
```diff
@@ -40,27 +41,24 @@ var addon = require( './dnannsumkbn2.native.js' );
...
-		offsetX += (N-1) * strideX;
-	}
-	if ( strideOut < 0 ) {
-		offsetOut += strideOut;
-	}
-	viewX = new Float64Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offsetX), x.length-offsetX ); // eslint-disable-line max-len
-	viewOut = new Float64Array( out.buffer, out.byteOffset+(out.BYTES_PER_ELEMENT*offsetOut), out.length-offsetOut ); // eslint-disable-line max-len
+
+	offsetX = minViewBufferIndex( N, strideX, offsetX );
+	offsetOut = minViewBufferIndex( N, strideOut, offsetOut );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/lib/ndarray.native.js]: The error you are getting stems from this line. I'll leave it to you to figure out why. Hint: are `x` and `out` the same length?

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/lib/ndarray.native.js]: I suggest studying L62 and L63 from the previous implementation. And I encourage you to not simply copy and paste but reason about the implementation.

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/src/addon.c (hunk 4 above)
Comment (vr-varad) [reply lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/src/addon.c]: Sorry for that actually for this pr I was constantly making changes so didn;t kept the indentation in mind, this won't happen again sorry @kgryte

**Comment by vr-varad** [comment]
@kgryte @Planeshifter I'm delighted to inform you that I've successfully resolved the error we encountered earlier. I want to extend my heartfelt gratitude to @kgryte for your invaluable guidance throughout this process.

I'm now ready to proceed with the review. Thank you once again for your assistance.

**Comment by vr-varad** [comment]
@kgryte @Planeshifter I'm delighted to inform you that I've successfully resolved the error we encountered earlier. I want to extend my heartfelt gratitude to @kgryte for your invaluable guidance throughout this process.

I'm now ready to proceed with the review. Thank you once again for your assistance.

**Review by Planeshifter — APPROVED** [review]
Thank you, @vr-varad!

**Pull request merged**